subsys/bluetooth/services/ddfs.c

```diff
@@ -277,23 +280,24 @@ static void ctrl_point_ind(struct bt_conn *conn, uint8_t req_op, uint8_t status,
 		net_buf_simple_add_mem(&buf, data, len);
 	}
 
-	ddfs_inst.ind_params.attr = &ddf_svc.attrs[DDF_SVC_CTRL_POINT_ATTR_IDX];
+	ddfs_inst.ind_params.attr = attr;
 	ddfs_inst.ind_params.func = indicate_cb;
 	ddfs_inst.ind_params.destroy = indicate_destroy;
 	ddfs_inst.ind_params.data = ind;
 	ddfs_inst.ind_params.len = buf.len;
 
-	if (bt_gatt_indicate(conn, &ddfs_inst.ind_params) == 0) {
+	err = bt_gatt_indicate(conn, &ddfs_inst.ind_params);
+	if (err) {
+		LOG_ERR("Failed to send Control Point indication, err: %d", err);
+	} else {
 		ddfs_inst.indicating = true;
 	}
 }
 
 int bt_ddfs_distance_measurement_notify(struct bt_conn *conn,
 					const struct bt_ddfs_distance_measurement *measurement)
 {
-	if (!ddfs_inst.dm_notification_enabled) {
-		return -EACCES;
-	}
+	static const struct bt_gatt_attr *attr = &ddf_svc.attrs[DDF_SVC_DISTANCE_MEAS_ATTR_IDX];
```

Comment:
Can we use `bt_gatt_find_by_uuid`? 
Then `DDF_SVC_DISTANCE_MEAS_ATTR_IDX` will be unnecessary.
Also in other places.

Reply:
I wanted to use it but I don't know a call context of this function so there could be a race when two threads would execute this function first time, both thread could call then `bt_gatt_find_by_uuid` but the result will be the same even then. I decided to hardcode attr in the API functions. What do you think?

Reply:
Yes, race can occur at the first check. So hardcoded attr seems to be a better option.